File: 8_kyu/Contamination_1_String.go
package kata

import (
	"strings"
	"unicode/utf8"
)

// Replace all characters in the text by char.
// String char must contain one character.
// Returns empty string if text is empty or char is not one character.
func Contamination(text, char string) string {
	// use utf8 fpr a multi-byte character safe implementation.
	textLen := utf8.RuneCountInString(text)
	charLen := utf8.RuneCountInString(char)
	if textLen <= 0 || charLen != 1 {
		return ""
	}
	return strings.Repeat(char, textLen)
}

File: 8_kyu/Contamination_1_String.rs
// Retuns string of length text filled with character.
fn contamination(text: &str, character: &str) -> String {
    // character must be a string contaning 1 character.
    if character.chars().count() != 1 {
        return String::new()
    }
    // Calculate number of chars, 
    let n = text.chars().count();
    // Return string with same number of chars as text.
    character.repeat(n)
}
